### ISSUE TITLE
Document map is sluggish and laggy.

### ISSUE THREAD

**Issue by leet0rz** [notepad-plus-plus/notepad-plus-plus#10046]
### Description of the Issue
Title, its slow, sluggish and laggy. Is it possible to improve?

### Steps to Reproduce the Issue
1. Use it.

### Expected Behavior
Not slow, laggy and sluggish.

### Actual Behavior
Slow, sluggish and laggy

### To add
Please also make the scrollbar on the right side dark on the new dark mode its almost impossible to see.

**Comment by leet0rz** [comment]
up

**Comment by leet0rz** [comment]
up

**Comment by mere-human** [comment]
Care to provide more information?
For example, files where the problem is visible and/or a video demonstrating the problem?

**Comment by leet0rz** [comment]
Any file with any text, by simply using the document map. I dont know what more to say. Here is a gif of how copy, laggy and glitchy it looks:
![recording(1)](https://user-images.githubusercontent.com/29175192/130089267-0cd9c7c2-c431-4448-865e-686e5d16eaa7.gif)

**Comment by felix-22** [comment]
I can't reproduce even on such big file like the 300k https://github.com/notepad-plus-plus/notepad-plus-plus/issues/8279#issuecomment-743696790

**Comment by leet0rz** [comment]
> 
> 
> Care to provide more information?
> For example, files where the problem is visible and/or a video demonstrating the problem?

Good enough?

**Comment by felix-22** [comment]
> leet0rz : Good enough?

To be honest, not really, fill the template properly, paste DEBUG info.

**Comment by leet0rz** [comment]
> 
> 
> > leet0rz : Good enough?
> 
> To be honest, not really, fill the template properly, paste DEBUG info, don't make developers' lives harder.

Notepad++ v8.1.3   (32-bit)
Build time : Aug 10 2021 - 00:32:53
Path : D:\Program Files (x86)\Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Enterprise (64-bit) 
OS Version : 2009
OS Build : 19043.1165
Current ANSI codepage : 1252
Plugins : AutoSave.dll mimeTools.dll NppAutoIndent.dll NppConverter.dll NppExec.dll NppExport.dll NppSnippets.dll Python Indent.dll 

This what you are looking for? I Did fill out the previous part and send you a gif. Why did you close the thread?

**Comment by felix-22** [comment]
> leet0rz : This what you are looking for?

Yeah.

> leet0rz : Why did you close the thread?

Nobody closed the thread, it's the other thread closed.

> leet0rz : I Did fill out the previous part properly as its a pretty simple issue and send you a gif. 

If it's so simple, why can't the problem be reproduced on our side? We can't fix a bug that doesn't occur on our side. It is not known whether the bug exists in the program or in the user's system. Many bugs turns out to be caused by the user's additional settings, plugins, apps running in the background (anti-virus etc), some changes in the operating system etc.

> leet0rz : Notepad++ v8.1.3 (32-bit)
> leet0rz : Plugins : AutoSave.dll mimeTools.dll NppAutoIndent.dll NppConverter.dll NppExec.dll NppExport.dll NppSnippets.dll Python Indent.dll

Users like you really make developers' life harder: https://github.com/notepad-plus-plus/notepad-plus-plus/issues/10460#issuecomment-907830678. 
You have a lot of additional plugins, changed settings and an old version of the program, even a monkey would be smart enough to make sure that the problem persists with the latest program version and default plugins before creating a report.

**Comment by leet0rz** [comment]
I mean it was a pretty simple issue to explain and fill in in the form, so I do not know what you were whining about in terms of filling it in, I also sent you the gif to show you what was happening. No, you make your own life hard with notepad++ as it looks way out of date and is just outperformed these days by other programs, lol. That is not a lot of plugins, that is minimal compared to many other users. A monkey would be smart enough to see if the problem still persists with the lastest version of the program? Are you serious? This was posted when that version was the latest version 😂 I just opened it again and updated and it is still sluggish and garbage in the document map, it also just looks awful compared to other programs, do better!

**Comment by felix-22** [comment]
> leet0rz : I mean it was a pretty simple issue to explain and fill in in the form, so I do not know what you were whining about in terms of filling it in, I also sent you the gif to show you what was happening

It seems like you're the one whining here, just because you were asked to fill it out properly.
Also, you already asked about it and I already answered you, this is the answer:

> > leet0rz : I Did fill out the previous part properly as its a pretty simple issue and send you a gif.
> 
> felix-22 : If it's so simple, why can't the problem be reproduced on our side? We can't fix a bug that doesn't occur on our side. It is not known whether the bug exists in the program or in the user's system, many bugs turns out to be caused by the user's additional settings, plugins, apps running in the background (anti-virus etc), some changes in the operating system etc.

so why asking twice? Which part of my answer do you not understand?

> leet0rz : No, you make your own life hard with notepad++ as it looks way out of date and is just outperformed these days by other programs, lol.

I didn't quite get your point, what are you whining about? 

> leet0rz : That is not a lot of plugins, that is minimal compared to many other users. 

The developers of N++ do not create external plugins, are not responsible for the bugs in them, and are not in the business of fixing them, nor do they have the interest or time to install X additional plugins to see if it is one of the plugins that is causing the problems. It's the reporter's job to make sure that the bug he is reporting isn't caused by him. Even a monkey knows that checking whether a bug occurs with the default settings, is standard procedure before reporting a bug.

> leet0rz : A monkey would be smart enough to see if the problem still persists with the lastest version of the program? Are you serious? This was posted when that version was the latest version you polka monkey I just opened it again and updated and it is still sluggish and garbage in the document map, it also just looks awful compared to other programs, do better!

> leet0rz : This was posted

I don't quite get your point, I wasn't complaining about the fact that you didn't paste the 8.1.4 debug log in your first comment, because how could I expect you to? I was complaining about the fact that you pasted the 8.1.3 debug log instead of the 8.1.4 debug log in your [last comment](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/10046#issuecomment-908753968), so what are you whining about? 

> leet0rz : you polka monkey

It is better to use arguments than to use name-calling, really.

**Comment by dail8859** [comment]
@felix-22 @leet0rz  Let's keep things civilized please. Some of the above comments were not needed.

**Comment by felix-22** [comment]
I appreciate the attempt to calm the situation down, but only people who have misbehaved should be called to order, so until you can prove to me that there was something wrong with my comments, and that it was my fault, I don't like being called out by nickname and put on the same level as the one who misbehaved.

**Comment by dail8859** [comment]
@felix-22

I'll be the first to admit that there are many times it is hard to help users with their issues. It is best to assume good intentions and not blame people for specifically making "developer's lives harder" or to imply they are not intelligent. Let's focus on the issue, not the individuals.

> Do you mean my comments as well?

Yes.

----

Let's keep things focused on the issue that was submitted.

**Comment by felix-22** [comment]
> dail8859  : It is best to assume good intentions and not blame people for specifically making "developer's lives harder"

1) It's a bit hypocritical that you say about not to blame, and you do the same and are blaming me and  breaking your own rules...
2) Such issues DO make developer's life harder, rarely they can be reproduced, more trouble than it's worth, as a proof an example of an issue you just closed for the same reason, https://github.com/notepad-plus-plus/notepad-plus-plus/issues/10465#issuecomment-908296300, see, even more of hypocrisy, users who create such issues should be made to understand clearly and firmly that they should follow the template, otherwise we will drown in such trash issues.

> dail8859 : or to imply they are not intelligent. 

I first requested him to fill out the template correctly, and then he started complaining unnecessarily that he doesn't need to fill properly the template, if an user is disrespectful to the template and to providing important information, developers can be disrespectful for the user, simply, so why blaming me?

**Comment by dail8859** [comment]
@leet0rz Can you try enabling word-wrap? `View > Word Wrap` make sure it is checked. See if that makes any difference.

**Comment by leet0rz** [comment]
@felix-22 Are you serious? It is so clear that you behaved unprofessionally in this thread for no reason and now you are trying to justify your behavior like a little child who has clearly done wrong but cant accept his wrong-doings. You started the name calling, calling people monkeys for not upgrading to the latest version of the program when I was already on the latest version when I posted what I did, I cant be on a future version now can I? Use your brain for once and own up to you starting the name calling. You also started the whining about how I made the post in the first place, do not get it twisted. You are highly unprofessional and should really have no place in helping people on github at all. Now you are also trying to put down your fellow helper who called you out for acting like a child, disgusting.

**Comment by leet0rz** [comment]
> 
> 
> @leet0rz Can you try enabling word-wrap? `View > Word Wrap` make sure it is checked. See if that makes any difference.

Appreciate it, that did fix the document map lagging out and being sluggish however the file itself is still a bit choppy, do you know what causes this?

**Comment by dail8859** [comment]
> Appreciate it, that did fix the document map lagging out and being sluggish however the file itself is still a bit choppy, do you know what causes this?

I don't know anything else immediately but knowing word wrap helps can start pointing to potential causes.

**Comment by leet0rz** [comment]
> 
> 
> > Appreciate it, that did fix the document map lagging out and being sluggish however the file itself is still a bit choppy, do you know what causes this?
> 
> I don't know anything else immediately but knowing word wrap helps can start pointing to potential causes.

Thanks a lot, that did help make it a lot better than before. No more flickering.

**Comment by felix-22** [comment]
> leet0rz : Are you serious? It is so clear that you behaved unprofessionally in this thread for no reason and now you are trying to justify your behavior like a little child who has clearly done wrong but cant accept his wrong-doings. 
> leet0rz  : You are highly unprofessional and should really have no place in helping people on github at all.

Are you serious? You must be kidding...actually this is a perfect description of your behavior, not mine...

> leet0rz : and now you are trying to justify your behavior like a little child who has clearly done wrong but cant accept his wrong-doings. 

You are the one who behaves exactly like this, and not me, I'm the one with arguments, and I don't see any arguments in your comment, only personal taunts and turning everything upside down, this is called trolling.

> leet0rz  :  You started the name calling, calling people monkeys for not upgrading to the latest version of the program when I was already on the latest version when I posted what I did, I cant be on a future version now can I?

Nope, you were not on the latest version, you provided DEBUG info from 8.1.3, and the newest is 8.1.4...also, you had a lot of extra plugins and altered settings.

> leet0rz :  You also started the whining about how I made the post in the first place, do not get it twisted.

I rightly pointed out to you that you are making it difficult to solve the issue with such reports, do not get it twisted just because I rightly pointed it out to you.

> leet0rz :  Now you are also trying to put down your fellow helper who called you out for acting like a child, disgusting.

Stop trolling and maniulation, he called out both of us, not just me. And thus he made me equal to you, which I absolutely cannot agree on, and that was the only reason why I continued talking to him, to make him aware of it. Your comments are full of personal taunts and fact manipulations, where my comments are argumentative. You caused the whole problem with your garbage bug report, so you were called to order, and now you're crying because you can't accept that someone made a valid point to you.

> > dail8859 : @ leet0rz Can you try enabling word-wrap? `View > Word Wrap` make sure it is checked. See if that makes any difference.
> 
> leet0rz : Appreciate it, that did fix the document map lagging out and being sluggish however the file itself is still a bit choppy, do you know what causes this? Thanks a lot, that did help make it a lot better than before. No more flickering.

`Word-wrap` is enabled by default you noob, you disabled it. And then you made a ticket without first checking if the problem still occurs on the default settings. So unprofessional, and you call others unprofessional? Grow up kiddie. You are out of arguments (you never had them anyway) Stop wasting others' time and spoiling their nerves with your unprofessional behavior. **Game Over.**

Just don't cry again and don't complain. Bye.

**Comment by mlatushko** [comment]
### Expected Behavior
+ Need a pre-looking scrollbar (3 items before the end of the column).

### Expected Behavior
+ Need to focus the scrollbar over the same file after re-size program's window.
### Actual Behavior
Now it looses the file and resets view to the top.

**Comment by leet0rz** [comment]
@felix-22 There is something seriously wrong with you. Referring you back to my previous message as it fits you perfectly, remove yourself I do not need your help.

**Comment by felix-22** [comment]
> leet0rz : My version was not even 8.1.3 when I made this post in the first place as it was not available yet

You were already complaining about it, and I already defended and adressed it by saying it was not about the first comment:

>  felix-22 : https://github.com/notepad-plus-plus/notepad-plus-plus/issues/10046#issuecomment-909071450 :
> > leet0rz :  leet0rz : A monkey would be smart enough to see if the problem still persists with the lastest version of the program? Are you serious? This was posted when that version was the latest version you polka monkey
> 
>  I don't quite get your point, I wasn't complaining about the fact that you didn't paste the 8.1.4 debug log in your first comment, because how could I expect you to? I was complaining about the fact that you pasted 8.1.3 debug log instead of 8.1.4 debug log in your [last comment](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/10046#issuecomment-908753968), so what are you whining about?

And then I asked you "what are you whining about? You didn't reply...and then you're now complaining again about the same , despite you were given the explanations, so I've quoted my reply again...and I'm asking again: Which words of my reply don't you still understand? Simply: it was not about the first comment, it was about the other  comment, it was about [this comment](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/10046#issuecomment-908753968), you provided DEBUG of 8.1.3 instead of 8.1.4 Got it finally?

<hr>

> leet0rz : 8.1.3 was just the version I was on while asked for the debug, 

You said something different before, you said you were on the newest version, and now you're saying something different that you were on the version you were on while asked for the debug...a proof:

> > leet0rz : for not upgrading to the latest version of the program when I was already on the latest version when I posted what I did, I cant be on a future version now can I?
> 
> felix-22 : Nope, you were not on the latest version, you provided DEBUG info from 8.1.3, and the newest is 8.1.4...also, you had a lot of extra plugins and altered settings.

It was about to provide DEBUG from the latest version version, and not the version you were on while asked for the debug, what is point of providing DEBUG from 8.1.3 given the bug could be already fixed in 8.1.4? Like I already said before: Before reporting the issue, a standard procedure is to try to reproduce the bug on the newest version of the program, and not only on the version you are using at the time you are reporting the bug...the same apply to settings/plugins, you should try to reproduce on default setting/plugins...

<hr>

> leet0rz : There is something seriously wrong with you, go get help or something. Referring you back to my previous message as it fits you perfectly and all the information there is correct.

Nope, I've just proved that you are the one that there is something wrong with.

> leet0rz : use your brain next time you Polish monkey. Remove yourself, I do not need your "help" as you are just malicious and not helpful at all.

Nope, I've just proved, that you are the one who "should start using your brain" and is "a monkey".

> leet0rz : I do not need your "help" as you are just (...) and not helpful at all.

- nope, it's exactly the opposite, looking at my contribution to N++: [LINK 1](https://github.com/notepad-plus-plus/notepad-plus-plus/issues?q=is%3Aissue+commenter%3Afelix-22) , and yours [LINK 2](https://github.com/notepad-plus-plus/notepad-plus-plus/issues?q=is%3Aissue+commenter%3Aleet0rz+) it's clear who is more helpful here, your only help/contribution is a single garbare bug report mostly caused by settings on your side...go a walk.
- also like I already said you were already complaining about it, and I already defended and adressed it already,:

> leet0rz : You are (...) and should really have no place in helping people on github at all.
> 
> felix-22 First, you don't help us yourself, and then you tell morals about helping? Wait...isn't that a hypocrisy?

Since I provided a reply, I don't know why are you complaining again about the same, what don't you understand in the previous reply? I can just reply the same again: It's hard to help you when you don't want to help us first. Got it now? 

<hr>

> leet0rz : Remove yourself,

Given how classy I've been defending myself against all your accusations and childish insults, I would advise that you be the one to "remove yourself." It would be good if you would limit yourself to arguments and not call out names like a nervous little child, because it's so "unprofessional", so "malicious", so "disgusting"...

Visit a doctor, or stop taking drugs.

**Comment by felix-22** [comment]
> @donho Can we please block @felix-22 from this organization?

Are you serious? You kust be kidding... Why do you want to block me? Looking at the whole conversation, it's the leet0rz who recently is insulting me and trolling all the time, while I'm replying to him with arguments and no name calling...so if you want to block anyblody, it's leet0rz who meets the criteria...go a walk.

**Comment by felix-22** [comment]
@dail8859 I must admit that as far I see it's leet0rz who is insulting mostly...

**Comment by keczuppp** [comment]
@dail8859 I have to admit too that from what I see, it's leet0rz who is mostly insulting...so what's is this all about?

**Comment by dail8859** [comment]
@keczuppp I never said felix was the only one at fault, but by far seems to be the one provoking and continuing the matter.

**Comment by ArkadiuszMichalski** [comment]
@leet0rz

Did you finally check this in the last version 8.1.4 portable, without any plugins and fresh config? You got such advice, but somehow I don't see you follow it (unless I'm wrong, point where the answer is). If you still have this problem on that version then give a file for which you have this problem, because no one will extract it from the gif.

**Comment by donho** [comment]
@dail8859 
Indeed. Even though I don't like users who don't mind to provide complete information, the behaviour of @felix-22 is not acceptable.
I don't check every issue, so feel free to tag me if such situation happens again.

@felix-22 Do the support cordially, then you're welcome. Otherwise just stop.

**Comment by donho** [comment]
@leet0rz 
> Notepad++ v8.1.3 (32-bit)
> Build time : Aug 10 2021 - 00:32:53
> Path : D:\Program Files (x86)\Notepad++\notepad++.exe
> Command Line :
> Admin mode : OFF
> Local Conf mode : OFF
> Cloud Config : OFF
> OS Name : Windows 10 Enterprise (64-bit)
> OS Version : 2009
> OS Build : 19043.1165
> Current ANSI codepage : 1252
> Plugins : AutoSave.dll mimeTools.dll NppAutoIndent.dll NppConverter.dll NppExec.dll NppExport.dll NppSnippets.dll Python Indent.dll
> 
> This what you are looking for? I Did fill out the previous part properly as its a pretty simple issue and send you a gif. Why did you close the thread?

The debug info is as important as the bug description. A fix of performance regression has been fixed in v8.1.4, Also, the plugin list show the plugins you use. They should be excluded for your test to determinate if it's come from Notepad++ or plugins. 
So the debug information does matter.

**Comment by felix-22** [comment]
> dail8859 : > donho Can we please block @ felix-22 from this organization? I think only you have access to do so and this argument is getting out of hand.
> donho : @ dail8859 Indeed. Even though I don't like users who don't mind to provide complete information, the behaviour of @ felix-22 is not acceptable.
> donho : @ felix-22 Do the support cordially, then you're welcome. Otherwise just stop.

Go fuck your mother, then you're welcome. Reporting garbare and incomplete issues, and unnecessary complaining when asked to complete the template, is not cordial either, this is a plague, and plagues are stamped out, the guy deserved to be told about the monkeys. And I didn't deserve that he would keep calling me names all the time in all of his comments. You didn't mind when he called me names, I hope you didn't mind when I told you to fuck your mother.

**Comment by leet0rz** [comment]
@felix-22 No, what I wrote about the version last time was correct, not whatever you are trying to say. Completely delusional.

You are the one who started the name calling, calling people monkey in this thread, absolutely nothing happened before that as we all can read above, YOU started. You are highly unprofessional and have no place helping people anywhere. For you to just start calling people monkeys in a help thread tells me more about you than anything, go get your much needed mental care as I do not have time to deal with your psychological issues. Garbage places produce garbage people and I am blocking that garbage now :)

**Comment by leet0rz** [comment]
> 
> 
> @leet0rz
> 
> Did you finally check this in the last version 8.1.4 portable, without any plugins and fresh config? You got such advice, but somehow I don't see you follow it (unless I'm wrong, point where the answer is). If you still have this problem on that version then give a file for which you have this problem, because no one will extract it from the gif.

Yeah, I did check it on the 8.1.4 right away. Its the same, the word wrap did help quite a bit. Maybe I've turned it off with some hotkey if it is default as I do not remember ever turning that off. I have issues with all files, the one was just an example I took from a random song I googled. It would happen entirely the same with any other text copied.

Right now when I am using the docu map the main screen with text is choppy, the docu map itself is fine now after word wrap.

**Comment by leet0rz** [comment]
> 
> 
> @leet0rz
> 
> > Notepad++ v8.1.3 (32-bit)
> > Build time : Aug 10 2021 - 00:32:53
> > Path : D:\Program Files (x86)\Notepad++\notepad++.exe
> > Command Line :
> > Admin mode : OFF
> > Local Conf mode : OFF
> > Cloud Config : OFF
> > OS Name : Windows 10 Enterprise (64-bit)
> > OS Version : 2009
> > OS Build : 19043.1165
> > Current ANSI codepage : 1252
> > Plugins : AutoSave.dll mimeTools.dll NppAutoIndent.dll NppConverter.dll NppExec.dll NppExport.dll NppSnippets.dll Python Indent.dll
> > This what you are looking for? I Did fill out the previous part properly as its a pretty simple issue and send you a gif. Why did you close the thread?
> 
> The debug info is as important as the bug description. A fix of performance regression has been fixed in v8.1.4, Also, the plugin list show the plugins you use. They should be excluded for your test to determinate if it's come from Notepad++ or plugins.
> So the debug information does matter.

Is it possible to disable the plugins without uninstalling them or maybe I should virtualbox a notepad++ without anything to see if it works there? I would like to avoid deleting some of those settings.

**Comment by keczuppp** [comment]
>  leet0rz : https://github.com/notepad-plus-plus/notepad-plus-plus/issues/10046#issuecomment-912167971

That he called you a monkey once or twice does not make him a murderer and does not give you the right to constantly angry calling names at him in every comment you make, you are overreacting, it's time for you to finally stop as well, strange that the moderators pretend not to see this, anyway just let it go.

> leet0rz :  I should virtualbox a notepad++ without anything to see if it works there?

Exacly, you should have been done this even before reporting the issue..., also maybe this is your GPU drivers broken, did you try to upgrade your GPU drivers? Or try changing / resetting graphic settings in graphic panel? Or run Windows in Safe Mode? This may look like some rendering glitch causing the lag.

**Comment by dail8859** [comment]
@leet0rz 
> Is it possible to disable the plugins without uninstalling them

Yes, run Notepad++ from the command line and pass in the `-noPlugin` argument.

**Comment by donho** [comment]
> Go f..k your mothers you both, then you're welcome.

FYI, I've block felix-22.

**Comment by garry-ut99** [comment]
> keczuppp : https://github.com/notepad-plus-plus/notepad-plus-plus/issues/10046#issuecomment-912452181 : strange that the moderators pretend not to see this

Yeah, as seen, they manipluate a lot, as they only quote felix-22's vocabulary and only ban felix-22, while at the same time, for unknown reasons, they completely ignore leet0rz's vocabulary and refuse to ban leet0rz, even despite leet0rz offends 2 or 3 times more, even a blind man could see that... The most funny thing is that the offensive word donho just quoted, was provoked by their manipulation I just descirbed...

>> donho :  felix-22 : Go f..k your mothers you both, then you're welcome.
> 
> FYI, I've blocked felix-22.

Are you serious? Then now do the same with leet0rz, quote leet0rz's vocabulary and ban leet0rz...

**Comment by leet0rz** [comment]
I have a feeling @keczuppp and @garry-ut99 are just his alt accounts as they are both trying really hard to do damage control for felix-22, even though he started the name-calling for no reason. I am blocking both of them too.

**Comment by chcg** [comment]
The issue is happening to the document map if the length of the last visible line on screen is changing from lines with long input to  short/empty ones or vice versa.
It seems the document map has some logic to move to the end of a long line.

**Comment by itteerde** [comment]
Seems odd to me that for long files the Document Map is not the whole document, so scrolling on the bottom of the Document Map it just starts scrolling, to me defeating the purpose of the whole Document Map.
